examples/kafka/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"os"
+	"time"
+
+	kafka "github.com/segmentio/kafka-go"
+)
+
+func main() {
+	// Get the connection info from the ENV vars
+	host := os.Getenv("HOST")
+	port := os.Getenv("PORT")
+	topic := os.Getenv("TOPIC")
+
+	// And create a new kafka reader
+	reader := kafka.NewReader(kafka.ReaderConfig{
+		Brokers:  []string{host + ":" + port},
+		Topic:    topic,
+		MinBytes: 10e3,
+		MaxBytes: 10e6,
+	})
+	defer reader.Close()
+
+	// Open the /pfs/out pipe with write only permissons (the pachyderm spout will be reading at the other end of this)
+	// Note: it won't work if you try to open this with read, or read/write permissions
+	out, err := os.OpenFile("/pfs/out", os.O_WRONLY, 0644)
+	if err != nil {
+		panic(err)
+	}
+	defer out.Close()
+
+	// this is the file loop
+	for {
+		if err := func() error {
+			tw := tar.NewWriter(out)
+			defer tw.Close()
+
+			var b bytes.Buffer
+
+			// we'll use a ticker so that our spout will write files containing the kafka messages it consumed in 20ms intervals
+			tick := time.NewTicker(20 * time.Millisecond)
+			defer tick.Stop()
+			// if we wanted one file per message, we could do that by removing the for loop and the select code,
+			// so we would just read the message, and write the header and the message, and return
+
+			// this is the message loop
+			for {
+				select {
+				default:
+					// read a message and write it to the buffer
+					m, err := reader.ReadMessage(context.Background())
+					if err != nil {
+						panic(err)
+					}
+					b.Write(m.Value)
+				case <-tick.C:
+					// on each tick
+					// if there isn't anything in the buffer, we'll skip this interval
+					if b.Len() == 0 {
+						continue
+					}
+					// give this a unique name
+					name := topic + time.Now().Format(time.RFC3339Nano)
+					// write the header
+					if err = tw.WriteHeader(&tar.Header{
+						Name: name,
+						Mode: 0600,
+						Size: int64(b.Len()),
+					}); err != nil {
+						return err
+					}
+					// and the buffer
+					if _, err = b.WriteTo(tw); err != nil {
+						return err
+					}
+					// and reset the buffer
+					b.Reset()
+					return nil // this takes us back to the file loop
+				}
+			}
```

Comment:
This part seems kind of weird to me. I'm not sure I see why we need the select and the ticker. I was expecting it to look something like this, I'm probably missing why this won't work:

```golang
for {
    m, _ := reader.ReadMessage(...)
    if err := func() error {
        tw := tar.NewWriter(out)
        defer tw.Close()
        tw.WriteHeader(...)
        tw.Write(...)
    }(); err != nil {
	panic(err)
    }
}
```

Reply:
That would work fine too, I did it with the ticker both to test that the current setup was working robustly, and also because I thought that batching messages by time-intervals might be something people would want to do, and so this could be a helpful guide.

Reply:
I see, that makes a lot of sense. I think even though people probably will want to batch things for production kafka spouts for a demo we're probably better off doing the simplest thing possible so they can grok how to connect the dots between kafka and Pachyderm and then build batching and stuff on top of that.

Reply:
The simplified example is incompatible for the test I wrote, so I just included it as a commented out code block. 

I'd rather leave it as-is for now, and worry more about the exposition in the follow-up docs with @pappasilenus

Reply:
How hard would it be to change the test to be compatible with the simplified code? I feel like it's going to be even more confusing for users to have the more complicated code and then a commented out simpler version. @pappasilenus, would be good to hear your thoughts re the exposition of this. But my guess would be that the exposition is going to be easier of the code it's exposing is simpler. I also think there's going to be a decent number of people that just skip right to the code and don't read the exposition around it, so it's probably worth it to make that code as simple as it possibly can be.